modules/Rock.cmake

```diff
@@ -966,6 +976,69 @@ function(rock_gtest TARGET_NAME)
 
     rock_setup_gtest_test(${TARGET_NAME} ${GMOCK_DIR} ${GTEST_DIR})
     rock_add_test(${TARGET_NAME} "${__rock_test_parameters}")
+
+    if (ROCK_CLANG_STYLING_CHECK_ENABLED OR ROCK_CLANG_LINTING_CHECK_ENABLED)
+        get_target_property(test_files ${TARGET_NAME} SOURCES)
+        list(TRANSFORM test_files PREPEND "${PROJECT_SOURCE_DIR}/test/")
+        list(TRANSFORM library_files PREPEND "${PROJECT_SOURCE_DIR}/src/")
+        set(analyzed_files "${test_files};${library_files}")
+        list(FILTER analyzed_files EXCLUDE REGEX ".*gmock|gtest.*")
+
+        if (ROCK_CLANG_STYLING_CHECK_ENABLED)
+            rock_setup_styling_check(${TARGET_NAME} "${analyzed_files}")
+        endif()
+        if (ROCK_CLANG_LINTING_CHECK_ENABLED)
+            rock_setup_linting_check(${TARGET_NAME} "${analyzed_files}")
+        endif()
+    endif()
+endfunction()
+
+function(rock_setup_styling_check TARGET_NAME analyzed_files)
+    # Setup clang-format styling report
+    message(STATUS "Setting up linting for ${TARGET_NAME}")
+    find_program(
+        clang_format_exec NAMES 
+        clang-format clang-format-10 clang-format-11 clang-format-12
+        clang-format-13 clang-format-14 clang-format-15 ${CLANG_FORMAT_EXECUTABLE}
+    )
+    if (clang_format_exec)
+        add_test(
+            NAME
+            clangformat
+            COMMAND
+            ${clang_format_exec}
+            -style=file:${clang_format_config_path}
```

Comment:
The convention is to use uppercase letters for external variables (i.e. CLANG_FORMAT_CONFIG_PATH)

I would also add a sanity check and do a FATAL_ERROR if the PATH variable is unset while the corresponding pass is enabled with the corresponding option.